### DISCUSSION TITLE
Setting locale and timezone in SULU CMS

### DISCUSSION THREAD

**Discussion by nlipkanov** [sulu/sulu#6846] [Q&A] [answered]
I am actually not sure if this is an issue, but here is my question:

Tinkering around with the SULU CMS, I came to this weird situation when setting the default locale for the SULU CMS to "de" (german)
the input fields of type "date" however do not change to that locale. I am using the SuluFormBundle for my forms by the way.
Is there anything else I need to configure besides the configuration for the localization from the SULU docs or did I miss something?

**Comment by alexander-schranz** [comment, marked as answer]
@nlipkanov If I remember correctly the input format for date/times will normally be based on your browser localization or operation system language. The library does not keep the Sulu locale in mind. So they don't change if you change the Sulu Language. So if your operation system / browser is in EN it will use EN based format.

**Reply by nlipkanov** [reply]
I have set my OS to german and my browser is already set to german but it still does not work.
Do I have to configure the SULU form bundle in some way to achieve my goal?

**Reply by alexander-schranz** [reply]
![Bildschirmfoto 2022-10-10 um 11 48 10](https://user-images.githubusercontent.com/1698337/194839529-44d70872-6957-4878-b560-e9c2f766dbeb.png)

In my case it correctly works and my navigator outputs the correct language, in the browser console:

```
(navigator.languages && navigator.languages.length) ? navigator.languages[0] : navigator.language 
```

Even my sulu user is EN it uses my system prefered format. I'm not sure if the underlaying INTL lbirary uses the navigator language or even can access the OS configured date format. As example in MACOS you can define the format somewhere in the settings of the operation system.

**Reply by nlipkanov** [reply]
Yes, I see the german language in my browser as well:

![image](https://user-images.githubusercontent.com/98106050/194840280-6215a6ba-45c8-4101-ba0a-1de0ba6e0ae7.png)


But my input date fields are still in english:

![image](https://user-images.githubusercontent.com/98106050/194840563-b3064ed1-46e5-4170-8160-a86d0196c1bd.png)

**Reply by alexander-schranz** [reply]
@nlipkanov okay I misinterpeted your question as I thought you are talking about Sulu and Sulu Admin. The frontend in case of the SuluFormBundle is a Symfony Form. You can create a form theme or you can create a FormExtension or overwrite the DateType service of the FormBundle if you want to change anything about that field for your Website.

**Reply by alexander-schranz** [reply]
Related docs: https://symfony.com/doc/current/reference/forms/types/date.html#format. And related file https://github.com/sulu/SuluFormBundle/blob/2.5/Dynamic/Types/DateType.php#L42 Defaults using `\IntlDateFormatter::LONG` which uses the server time formats.

**Reply by nlipkanov** [reply]
Is there any documentation and examples on how to create a FormExtension or overwrite the DateType Service of the FormBundle?

**Reply by alexander-schranz** [reply]
Symfony have documentation about overwrite a service or how to decorate service: https://symfony.com/doc/current/bundles/override.html#services-configuration. Also for form type extension https://symfony.com/doc/current/form/create_form_type_extension.html. Why form type extension can be critical when modifying a "General" FormType as you are changing the default behaviour of something which maybe somebodies else bundle expect that default behaviour. So I would more go int that way overwrite the sulu form bundle service.
